src/cobalt-host.c

```diff
@@ -85,6 +97,51 @@ const char *cobalt_host_get_app_exec(CobaltHost *host, GError **error) {
   return host->exec;
 }
 
+SemVer *cobalt_host_get_fp_version(CobaltHost *host, GError **error) {
+  if (!host->fp_version) {
+    g_autoptr(GKeyFile) key_file = g_key_file_new();
+    if (!g_key_file_load_from_file(key_file, FLATPAK_INFO_PATH, G_KEY_FILE_NONE, error)) {
+      g_prefix_error(error, "Loading Flatpak info: ");
+      return NULL;
+    }
+
+    g_autofree gchar *version_str = g_key_file_get_string(
+        key_file, FLATPAK_INFO_INSTANCE, FLATPAK_INFO_INSTANCE_FP_VERSION, error);
+    if (version_str == NULL) {
+      g_prefix_error(error, "Getting Flatpak version: ");
+      return NULL;
+    }
+
+    host->fp_version = g_new0(SemVer, 1);
+    g_autoptr(GRegex) regex = g_regex_new("^([0-9]+)\\.([0-9]+)\\.([0-9]+)", 0, 0, error);
```

Comment:
Bit of a nit: this can just be:

```suggestion
    g_autoptr(GRegex) regex = g_regex_new("^(\\d+)\\.(\\d+)\\.(\\d+)", 0, 0, error);
```

Reply:
I just don't like how confusing those `\\d` look